src/cube.js

```diff
@@ -220,16 +241,25 @@ class Cube {
             return this.storedMeasures[measureId].data;
         else if (this.computedMeasures[measureId] !== undefined) {
             const storeSize = this.storeSize;
-            const measureIds = this.storedMeasureIds;
-            const measures = measureIds.map(id => this.storedMeasures[id]);
-            const numMeasures = measures.length;
+            const measures = this.computedMeasures[measureId].variables({ withMembers: true });
+            const storedMeasures = measures.filter(measureId => !measureId.includes('__total'));
+            const measureTotals = measures
+                .filter(measureId => measureId.includes('__total'))
+                .reduce((acc, measureId) => {
+                    acc[measureId] = this.storedMeasures[measureId.replace('__total', '')].total;
+                    return acc;
+                }, {});
 
             // Fill result array
             const result = new Array(storeSize);
             const params = {};
+            Object.entries(measureTotals).forEach(([measureId, total]) => {
```

Comment:
instead of building the `measureTotals` first and the enumerate over its entries, i think we could save some time by adding needed numbers straight to the `params` object. eg, replace the `reduce` on line 248 with a `forEach`